media/js/src/question.tsx

```diff
@@ -1,45 +1,48 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { useNavigate } from 'react-router-dom';
-import { checkQuestion, raw2latex } from './utils';
-
-type Statement = {
-    pk: number;
-    question: string;
-    answer: string;
-    difficulty: number;
-    created_at: string;
-}
+import { checkQuestion, raw2latex, ExerciseData, Statement } from './utils';
 
 interface QuestionProps {
     statement: Statement;
     listNum: number;
     id: number;
     level: string;
+    idStr: string;
 }
 
 export const Question: React.FC<QuestionProps> = (
-    { statement, listNum, id, level }: QuestionProps) => {
+    { statement, listNum, id, level, idStr }: QuestionProps) => {
 
     const navigate = useNavigate();
 
     const exerciseSpaceHandler = () => {
-        const initData: QuestionProps = {
-            statement: statement,
-            listNum: listNum,
-            id: id,
-            level: level,
-        };
-        const qstate = [...new Array<QuestionProps>(initData)];
-        window.localStorage.setItem('exerciseSpace',
-            JSON.stringify(qstate));
-
-        navigate('/exercise/');
+        navigate('/exercise/' + idStr);
+    };
+
+    const [questionStatus, setQuestionStatus] = useState('');
+    const getQuestionStatus = () => {
+        try {
+            const data = JSON.parse(
+                window.localStorage.getItem(
+                    'question-' + idStr)) as ExerciseData[];
+            const questStatus = data[0].status;
+            setQuestionStatus(questStatus);
+        } catch (error) {
+            setQuestionStatus('none');
```

Comment:
Why the `none` string? Could `null` work better here?

Reply:
That was mostly for me to check that the information was passing through properly. Back to null.